### ISSUE TITLE
Other A11y: Setting-Save the World: Whole layout automatically closes after submitting form

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#74835]
Action Performed
Prerequisites:
- the user is signed

1. Using Chrome, open the URL - https://new.expensify.com/
2. Navigate to the 'Account' and press Enter > Navigate to the 'Settings - Save the world' and press Enter
3. Tab to the 'I know a teacher' and press Enter
4. Add input into the fields, Tab to the 'Let's do this' button, and press Enter
5. Note what happens.

Expected Result
After submitting the Teacher form, it should not automatically close the entire Account layout; the user should remain in the same context after submission
- A confirmation or success message (e.g., “Teacher details saved successfully”) should be displayed and announced
- Any navigation or layout change should only occur after explicit user confirmation

Actual Result
After submitting the Teacher form, the entire Account layout automatically closes; the user is unexpectedly taken out of the current view without any confirmation; this causes a loss of focus and context, particularly for screen reader and keyboard users

HTML:
<div class="css-g5y9jx r-13awgt0 r-1udh08x r-1pi2tsx r-12vffkv" style="background-color: rgba(0, 0, 0, 0);"><div class="css-g5y9jx r-13awgt0"><div class="css-g5y9jx r-184en5c r-12vffkv"></div><div class="css-g5y9jx r-13awgt0"><div class="css-g5y9jx r-13awgt0 r-18lll9h" data-testid="report-screen-7470147100835202" data-component="ScreenWrapperContainer" data-element="View" data-source-file="ScreenWrapperContainer.tsx"><div class="css-g5y9jx r-18lll9h r-1udh08x r-13awgt0" data-element="View" data-source-file="ScreenWrapperContainer.tsx" style="margin-top: 0px; padding-top: 0px;"><div class="css-g5y9jx r-13qz1uu r-1pi2tsx" data-element="KeyboardAvoidingView" data-source-file="ScreenWrapperContainer.tsx"><div class="css-g5y9jx r-13awgt0" data-component="PickerAvoidingView" data-element="View" data-source-file="index.js"><div class="css-g5y9jx r-13awgt0 r-13qz1uu r-1pi2tsx" data-element="View" data-source-file="index.tsx" webkitdirectory=""><div class="css-g5y9jx" data-component="OfflineWithFeedback" data-element="View" data-source-file="OfflineWithFeedback.tsx"><div class="css-g5y9jx"><div data-drag-area="true" class="css-g5y9jx r-qklmqi r-cv2ln6" data-element="View" data-source-file="HeaderView.tsx"><div class="css-g5y9jx r-6koalj r-wy61xf r-1777fci r-j2kj52" data-element="View" data-source-file="HeaderView.tsx"><div class="css-g5y9jx r-1awozwy r-18u37iz r-b5h31w"><div class="css-g5y9jx r-13awgt0 r-18u37iz r-1awozwy r-1wtj0ep"><button aria-label="#teachers-unite" role="button" data-selection-scrapper-hidden-element="true" data-tag="pressable" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21 r-18u37iz r-1awozwy r-13awgt0 r-9ffhgg r-lrvibr" type="button" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx"><div class="css-g5y9jx" data-component="OfflineWithFeedback" data-element="View" data-source-file="OfflineWithFeedback.tsx"><div class="css-g5y9jx"><div class="css-g5y9jx"><div class="css-g5y9jx r-y47klf r-1b7u577 r-633pao" data-testid="ReportActionAvatars-SingleAvatar"><div class="css-g5y9jx" style="overflow: hidden; border-radius: 8px;"><div class="css-g5y9jx r-1mlwlqe r-1udh08x r-417010 r-17gur6a" data-component="BaseImage" data-element="RNImage" data-source-file="BaseImage.tsx" style="height: 40px; width: 40px; background-color: rgb(26, 61, 50);"><div class="css-g5y9jx r-1niwhzg r-vvn4in r-u6sd8q r-1p0dtai r-1pi2tsx r-1d2f490 r-u8s1d r-zchlnj r-ipm5af r-13qz1uu r-1wyyakw r-4gszlv" style="background-image: url(&quot;https://d1wpcgnaa73g0y.cloudfront.net/25b34f34452b0f424128608dee7b24a69f4df669.png&quot;);"></div><img alt="" draggable="false" class="css-9pa8cd" src="[https://d1wpcgnaa73g0y.cloudfront.net/25b34f34452b0f424128608dee7b24a69f4df669.png"></div></div></div></div></div></div><div](https://d1wpcgnaa73g0y.cloudfront.net/25b34f34452b0f424128608dee7b24a69f4df669.png%22%3E%3C/div%3E%3C/div%3E%3C/div%3E%3C/div%3E%3C/div%3E%3C/div%3E%3Cdiv) class="css-g5y9jx r-13awgt0 r-eqz5dr fs-unmask"><div class="css-g5y9jx r-18u37iz r-9aw3ui r-1awozwy"><div dir="auto" class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-1iln25a fs-mask" data-testid="DisplayNamesWithTooltip" data-component="DisplayNamesWithToolTip" data-element="Text" data-source-file="DisplayNamesWithTooltip.tsx" style="color: rgb(231, 236, 233); font-size: 15px; text-align: left; font-family: &quot;Expensify Neue&quot;,<div>(...)<div>

Area issue was found
Settings-Save the World

Failed Checkpoint
WCAG 3.2.2 On Input
https://www.w3.org/WAI/WCAG22/Understanding/on-input

User Impact
This helps users with disabilities by making interactive content more predictable. Unexpected changes of context can be disorienting for users with visual disabilities or cognitive limitations. Individuals who are blind or have low vision may have difficulty knowing when a visual context change has occurred, such as a new window popping up. In this case, warning users of context changes in advance minimizes confusion when the user discovers that the back button no longer behaves as expected. Some individuals with low vision, with reading and intellectual disabilities, and others who have difficulty interpreting visual cues may benefit from additional cues in order to detect changes of context.

Suggested Resolution
Prevent automatic closure of the Account layout upon form submission; instead, add another button to close the layout after submitting the changes.
https://www.w3.org/WAI/WCAG22/Understanding/on-input

Environment(s)
Windows Windows 11 Home - Chrome

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6994066
**Issue reported by:** NA
**Slack conversation:**NA

https://github.com/user-attachments/assets/1644b8f6-9be9-40a6-a8be-9ff2479e01b7

<img width="1919" height="975" alt="Image" src="https://github.com/user-attachments/assets/b68c6425-430e-4887-8df6-1be54af59e7c" />

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by rushatgabhane** [comment]
@MelvinBot update the title of this issue to prepend "[HOLD]"

**Comment by MelvinBot** [comment]
Done — updated the issue title to prepend `[HOLD]`.

**Comment by rushatgabhane** [comment]
This issue was retested in **BFV cycle 544083 (Web)** on 2026-04-28 and is **Not Fixed**.

### Screenshots / Videos
- https://platform.applause.com/services/links/v1/external/84a99948120a6b7bae3570d87dff80199d768924e58a66155130ed8de5dcdbd7
- https://platform.applause.com/services/links/v1/external/0a16869beb2281e82a69e8351b5fa479499906b5502a42620f1ad4bfa9875c80

### Retest notes
Not fixed: After submitting the Teacher form, the entire Account layout automatically closes, and the user is taken out of the current view without any confirmation, causing loss of focus and context for screen reader and keyboard users.

**Comment by rushatgabhane** [comment]
not a valid bug https://expensify.slack.com/archives/C9YU7BX5M/p1778079534191459
